src/coins.cpp

```diff
@@ -814,7 +814,9 @@ bool CCoinsViewCache::UpdateSidechain(const CTransaction& tx, const CBlock& bloc
         scIt->second.sidechain.fixedParams.withdrawalEpochLength = cr.withdrawalEpochLength;
         scIt->second.sidechain.fixedParams.customData = cr.customData;
         scIt->second.sidechain.fixedParams.constant = cr.constant;
+        scIt->second.sidechain.fixedParams.certificateProvingSystem = cr.certificateProvingSystem;
```

Comment:
Note: the newly added field of ScCreation output must be verified in
`bool Sidechain::checkTxSemanticValidity(const CTransaction& tx, CValidationState& state)`
method to prevent transactions with invalid ScCreations to be included into mempool and/or state